In [ ]:
# import library
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [ ]:
# load data
customer_lifespan_data = pd.read_csv('https://docs.google.com/spreadsheets/d/1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI/export?format=csv')
daily_mkt_spendings = pd.read_csv('https://docs.google.com/spreadsheets/d/1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c/export?format=csv')
monthly_expenses = pd.read_csv('https://docs.google.com/spreadsheets/d/10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw/export?format=csv')
payroll = pd.read_csv('https://docs.google.com/spreadsheets/d/1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4/export?format=csv')
receipts_history = pd.read_csv('https://docs.google.com/spreadsheets/d/1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE/export?format=csv')

In [ ]:
customer_lifespan_data.head()

,Unnamed: 0,start_date,churn_date
0,1000,2021-11-15,2022-09-14
1,1001,2022-04-15,2023-02-16
2,1002,2022-10-30,2023-02-04
3,1003,2021-08-22,2023-02-07
4,1004,2021-08-23,2022-02-02


In [ ]:
daily_mkt_spendings.head()

,date,channel,spending
0,2023-01-01,Google Ads,784
1,2023-01-01,Facebook Ads,659
2,2023-01-01,LinkedIn Ads,729
3,2023-01-01,Twitter Ads,292
4,2023-01-02,Google Ads,935


In [ ]:
monthly_expenses.head()

,#,month,category,item,amount
0,1,2023-01,Server Costs,AWS Hosting,8000
1,2,2023-01,Server Costs,Google Cloud Storage,4000
2,3,2023-01,Software Licenses,Atlassian Jira,1200
3,4,2023-01,Software Licenses,Slack,800
4,5,2023-01,Software Licenses,Salesforce,1500


In [ ]:
payroll.head()

,month,department,employee_name,position,paid
0,2023-01,Sales,John Doe,Sales Manager,1500
1,2023-01,Sales,Jane Smith,Sales Associate,600
2,2023-01,Sales,Jim Brown,Sales Associate,700
3,2023-01,Sales,Laura Miller,Sales Associate,800
4,2023-01,Marketing,Alice Johnson,Marketing Manager,1650


In [ ]:
receipts_history.head()

,date,customer_id,receipt_amount,new_customer
0,2023-01-01,2653,67,1
1,2023-01-01,2731,271,1
2,2023-01-01,1277,231,0
3,2023-01-01,2094,107,0
4,2023-01-01,1314,416,0


In [ ]:
# Total spending includes: system, salary & marketing expenses
spending_system = monthly_expenses[(monthly_expenses['month']=='2023-03') & (monthly_expenses['item']=='Salesforce')]['amount'].sum()
salary = payroll[(payroll['month']=='2023-03') & (payroll['department'].isin(['Sales', 'Marketing']))]['paid'].sum()
mkt_expense = daily_mkt_spendings[(daily_mkt_spendings['date'] >= '2023-03-01') & (daily_mkt_spendings['date'] <= '2023-03-31')]['spending'].sum()
total_spending = spending_system + salary + mkt_expense

# New CX in March 2023
mar_new_customer = receipts_history[(receipts_history['date'] >= '2023-03-01') & (receipts_history['date'] <= '2023-03-31')
                                  & (receipts_history['new_customer'] == 1)]['customer_id'].nunique()

# CAC
CAC = total_spending/mar_new_customer
print(f'CAC: {CAC}')

CAC: 1213.968253968254


In [ ]:
# ARPU
mar_total_revenue = receipts_history[(receipts_history['date'] >= '2023-03-01' ) & (receipts_history['date'] <= '2023-03-31')]['receipt_amount'].sum()
mar_total_cx = receipts_history[(receipts_history['date'] >= '2023-03-01' ) & (receipts_history['date'] <= '2023-03-31')]['customer_id'].nunique()

ARPU = mar_total_revenue / mar_total_cx
print(f'ARPU: {ARPU}')

ARPU: 284.3595890410959


In [ ]:
# COG
service_fee = monthly_expenses[monthly_expenses['month'] == '2023-03']['amount'].sum()
salary = payroll[payroll['month'] == '2023-03']['paid'].sum()

COG = service_fee*0.6 + salary
print(f'COG: {COG}')

COG: 37584.0


In [ ]:
# Gross Margin
profit = mar_total_revenue - COG
gross_margin = profit / mar_total_revenue * 100
print(f'Gross Margin: {gross_margin}')

Gross Margin: 39.734804234460995


In [ ]:
# LTV
customer_lifespan = ( pd.to_datetime(customer_lifespan_data['churn_date']) - pd.to_datetime(customer_lifespan_data['start_date']) ).dt.days.mean()/30
LTV = ARPU * customer_lifespan * gross_margin
print(f'LTV: {LTV}')

LTV: 111196.95570776255


In [ ]:
# LTV / CAC
LTV_CAC = LTV / CAC
print(f'LTV / CAC: {LTV_CAC}')

LTV / CAC: 91.59791069023328


In [ ]:
# Summary
print(f'CAC: {CAC}')
print(f'ARPU: {ARPU}')
print(f'COG: {COG}')
print(f'Gross Margin: {gross_margin}')
print(f'LTV: {LTV}')
print(f'LTV / CAC: {LTV_CAC}')

CAC: 1213.968253968254
ARPU: 284.3595890410959
COG: 50040
Gross Margin: 39.734804234460995
LTV: 111196.95570776255
LTV / CAC: 91.59791069023328


### **Conclusion and Recommendations**



Based on the calculated unit economic, Streamline Pro demonstrates a strong and sustainable business model.

The company's CAC accounted for 1213 which is much higher than ARPU of 284 shows that the company incurred losses in the first few months. However, the LTV/CAC ratio of 91.6 indicates that the customer lifetime value significantly exceeds the acquisition cost, suggesting that marketing strategies are highly efficient. In addition, the gross margin of approximately 39.7% shows that the company maintains healthy profitability after covering production costs.

Overall, the results suggest that the company is operating with strong financial efficiency and has significant potential to scale its customer acquisition strategy.